redisinsight/ui/src/pages/browser/components/stream-details/StreamDetails/StreamRangeFilter.tsx

```diff
@@ -0,0 +1,190 @@
+    [firstEntryTimeStamp, lastEntryTimeStamp]
+  )
+
+  const minValRef = useRef<HTMLInputElement>(null)
+  const maxValRef = useRef<HTMLInputElement>(null)
+  const range = useRef<HTMLInputElement>(null)
+
+  const prevMaxValue = usePrevious({ max }) ?? { max: '' }
+
+  const resetFilter = useCallback(
+    () => {
+      dispatch(updateStart(firstEntryTimeStamp.toString()))
+      dispatch(updateEnd(lastEntryTimeStamp.toString()))
+    },
+    [firstEntryTimeStamp, lastEntryTimeStamp]
+  )
+
+  useEffect(() => {
+    if (maxValRef.current) {
+      const minPercent = getPercent(startNumber)
+      const maxPercent = getPercent(+maxValRef.current.value)
+
+      if (range.current) {
+        range.current.style.left = `${minPercent}%`
+        range.current.style.width = `${maxPercent - minPercent}%`
+      }
+    }
+  }, [startNumber, getPercent])
+
+  useEffect(() => {
+    if (minValRef.current) {
+      const minPercent = getPercent(+minValRef.current.value)
+      const maxPercent = getPercent(endNumber)
+
+      if (range.current) {
+        range.current.style.width = `${maxPercent - minPercent}%`
+      }
+    }
+  }, [endNumber, getPercent])
+
+  useEffect(() => {
+    if (start === '') {
+      dispatch(updateStart(firstEntryTimeStamp.toString()))
+    }
+  }, [min])
+
+  useEffect(() => {
+    if (end === '') {
+      dispatch(updateEnd(lastEntryTimeStamp.toString()))
+    }
+  }, [max])
+
+  useEffect(() => {
+    if (max && prevMaxValue && endNumber === getTimestampFromId(prevMaxValue.max)) {
+      dispatch(updateEnd(getTimestampFromId(max).toString()))
+    }
+  }, [prevMaxValue])
+
+  useEffect(() => {
+    if (start && end) {
+      dispatch(fetchStreamEntries(
+        key,
+        SCAN_COUNT_DEFAULT,
+        sortedColumnOrder,
+        false
+      ))
+    }
+  }, [start, end])
+
+  if (start === '' && end === '') {
+    return (
+      <div className={styles.rangeWrapper}>
+        <div style={{ left: '30px', width: 'calc(100% - 56px)' }} className={styles.sliderTrack} />
+      </div>
+    )
+  }
+
+  if (start === end) {
+    return (
+      <div className={styles.rangeWrapper}>
+        <div style={{ left: '30px', width: 'calc(100% - 56px)' }} className={styles.sliderRange}>
+          <div className={styles.sliderLeftValue}>{getFormatTime(start)}</div>
+          <div className={styles.sliderRightValue}>{getFormatTime(end)}</div>
+        </div>
+      </div>
+    )
+  }
+
+  return (
+    <>
+      <div className={styles.rangeWrapper}>
+        <input
+          type="range"
+          min={firstEntryTimeStamp}
+          max={lastEntryTimeStamp}
+          value={startNumber}
+          ref={minValRef}
+          onChange={(event) => {
+            const value = Math.min(+event.target.value, endNumber - 1)
+            dispatch(updateStart(value.toString()))
+            event.target.value = value.toString()
+          }}
+          className={`${styles.thumb} ${styles.thumbZindex3}`}
```

Comment:
please, add "data-testid" for tests for each input and divs with different texts